File: leetcode/js/find-most-frequent-vowel-and-consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function (s) {
  const vowels = new Map([
    ["a", 0],
    ["e", 0],
    ["i", 0],
    ["o", 0],
    ["u", 0],
  ]);
  const consonants = new Map();

  for (const c of s) {
    if (vowels.has(c)) {
      vowels.set(c, vowels.get(c) + 1);
    } else {
      const val = consonants.get(c);

      consonants.set(c, val ? val + 1 : 1);
    }
  }

  const maxVowelCount = Math.max(...vowels.values());
  const maxConsonantsCount = consonants.size
    ? Math.max(...consonants.values())
    : 0;

  return maxVowelCount + maxConsonantsCount;
};

File: leetcode/python/find-most-frequent-vowel-and-consonant.py
from collections import defaultdict, Counter

class Solution:
    def maxFreqSum(self, s: str) -> int:
        vowels = defaultdict(int)
        consonants = defaultdict(int)

        for c in s:
            if c in ['a', 'e', 'i', 'o', 'u']:
                vowels[c] += 1
            else:
                consonants[c] += 1
        
        max_vowels_count = max(vowels.values()) if vowels else 0
        max_consonants_count = max(consonants.values()) if consonants else 0
        
        return max_vowels_count + max_consonants_count

class Solution1:
    def maxFreqSum(self, s: str) -> int:
        vowels = defaultdict(int)
        consonants = defaultdict(int)
        max_v = 0
        max_c = 0

        for c in s:
            if c in "aeiou":
                vowels[c] += 1
                max_v = max(max_v, vowels[c])
            else:
                consonants[c] += 1
                max_c = max(max_c, consonants[c])
        
        return max_v + max_c
    
class Solution2:
    def maxFreqSum(self, s: str) -> int:
        counter = Counter(s)
        max_v = max((counter[c] for c in "aeiou"), default=0)
        max_c = max((v for c, v in counter.items() if c not in "aeiou"), default=0)

        return max_v + max_c